modules/geo-layers/src/tile-layer/utils.js

```diff
@@ -145,3 +145,10 @@ export function getTileIndices({viewport, maxZoom, minZoom, zRange, extent, tile
     ? getOSMTileIndices(viewport, z, zRange, extent || DEFAULT_EXTENT)
     : getIdentityTileIndices(viewport, z, extent || DEFAULT_EXTENT);
 }
+
+/**
+ * Returns true if s is a valid URL template
+ */
+export function isURLTemplate(s) {
+  return Boolean(s.match(/(?=.*{z})(?=.*{x})(?=.*({y}|{-y}))/));
```

Comment:
- Use `regex.test`
- Curly brackets should be escaped
- Any reason to require all three?

Reply:
Good point with regex.test, I've just done it.

We need the three because a URL template requires the three. I've added {-y} because it's used at  https://github.com/visgl/deck.gl/blob/8.3-release/modules/geo-layers/src/tile-layer/utils.js#L42-L46 (to give support to TMS).

Are you sure we need to escape the curly brackets? Check [here](https://regexr.com/5eeq9)